viewing_party/main.py

```diff
@@ -0,0 +1,271 @@
+    """
+
+    friends_movies = []
+    for friend in user_data["friends"]:
+        for movie in friend["watched"]:
+            if movie not in friends_movies:
+                friends_movies.append(movie)
+    return friends_movies
+
+def get_unique_watched(user_data):
+    """Determines which movies the user has watched, but none of their friends
+    have watched.
+
+    Args:
+        user_data (dict): Represents the user's data, with their
+            ``watched`` list of movies and ``friends`` list.
+
+    Returns:
+        user_unique_movies (list): a list of dictionaries, that 
+            represents a list of movies.
+    """
+
+    friends_movies = get_friends_movies(user_data)
+    user_unique_movies = []
+
+    for movie in user_data["watched"]:
+        if movie not in user_unique_movies and movie not in friends_movies:
+            user_unique_movies.append(movie)
+    return user_unique_movies
+
+def get_friends_unique_watched(user_data):
+    """Determines which movies at least one of the user's friends have watched, but the user has not watched.
+
+    Args:
+        user_data (dict): Represents the user's data, with their
+            ``watched`` list of movies and ``friends`` list.
+
+    Returns:
+        user_must_see (list): A list of dictionaries, that 
+        represents a list of movies.
+    """
+
+    friends_movies = get_friends_movies(user_data)
+    user_must_see = []
+
+    for movie in friends_movies:
+        if movie not in user_data["watched"] and movie not in user_must_see:
+                user_must_see.append(movie)
+    return user_must_see
+
+# test_wave_04
+
+def get_available_recs(user_data):
+    """Determines a list of movie recommendations based in movies that the user didn't watch and their ``subscriptions`` in streaming services.
+
+    Args:
+        user_data (dict): Represents the user's data, with their
+            ``watched`` list of movies, a ``friends`` list, and 
+            their ``subscriptions``.
+
+    Returns:
+        recommendations (list): A list of movies recommended by 
+            friends.
+    """
+
+    subscriptions = user_data["subscriptions"]
+    friends_movies = get_friends_movies(user_data)
+    recommendations = []
+
+    for movie in friends_movies:
+        if movie["host"] in subscriptions and movie not in recommendations:
+            recommendations.append(movie)
+    return recommendations
+
+# test_wave_05
+
+def get_new_rec_by_genre(user_data):
+    """Determines a list of recommended movies considering the user's most frequently watched movie ``genre``.
+
+    Args:
+        user_data (dict): Represents the user's data, with their
+            list of movies, favorite movie genre, and a ``friends``
+            list.
+
+    Returns:
+        new_recs (list): A list of recommended movies based in
+            the user's favorite genre.
+    """
+
+    friends_movies = get_friends_movies(user_data)
+    fav_genre = get_most_watched_genre(user_data)
```

Comment:
Great work using the previous functions as helper function to write clear, concise code